src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/RuntimeNodeWriter.cs

```diff
@@ -110,33 +85,32 @@ public override void WriteCSharpCode(CodeRenderingContext context, CSharpCodeInt
             return;
         }
 
-        IDisposable linePragmaScope = null;
-        if (node.Source != null)
-        {
-            linePragmaScope = context.CodeWriter.BuildLinePragma(node.Source.Value, context);
-            context.CodeWriter.WritePadding(0, node.Source.Value, context);
-        }
+        WriteCSharpChildren(node.Children, context);
+        context.CodeWriter.WriteLine();
+    }
 
-        for (var i = 0; i < node.Children.Count; i++)
+    private static void WriteCSharpChildren(IntermediateNodeCollection children, CodeRenderingContext context)
+    {
+        for (var i = 0; i < children.Count; i++)
         {
-            if (node.Children[i] is IntermediateToken token && token.IsCSharp)
+            if (children[i] is IntermediateToken token && token.IsCSharp)
             {
-                context.AddSourceMappingFor(token);
-                context.CodeWriter.Write(token.Content);
+                using (context.CodeWriter.BuildEnhancedLinePragma(token.Source, context))
+                {
+                    if (!context.Options.UseEnhancedLinePragma)
```

Comment:
Is this condition ever true? Seems that we would still emit enhanced pragma in the `BuildEnhancedLinePragma` above even if `context.Options.UseEnhancedLinePragma` would be `false`.

Reply:
No, the `BuildEnhancedLinePragma` call checks the condition too, and falls back to a regular pragma if the condition isn't met.

Reply:
I see, I got confused by hardcoded argument `useEnhancedLinePragma: true`, but the check is nested more deeply.

So don't we want to have this `if (!context.Options.UseEnhancedLinePragma) context.CodeWriter.WritePadding(0, token.Source, context);` in all the other places where we do `using (context.CodeWriter.BuildEnhancedLinePragma(token.Source, context))` ? Shouldn't it be even part of the `BuildEnhancedLinePragma` method?

Reply:
Good call, moved it into `BuildEnhancedLinePragma`. Actually fixed an issue where we were writing padding incorrectly for `@page` directives.